src/main/java/hudson/remoting/ProxyWriter.java

```diff
@@ -98,7 +103,7 @@ public void write(int c) throws IOException {
         write(new char[]{(char)c},0,1);
     }
 
-    public void write(char[] cbuf, int off, int len) throws IOException {
+    public synchronized void write(char[] cbuf, int off, int len) throws IOException {
```

Comment:
Are we sure this will not introduce risk of deadlock?

Reply:
Yes, it makes sense to reimplement it